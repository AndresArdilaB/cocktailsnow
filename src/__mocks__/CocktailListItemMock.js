const CocktailListItemMock = {
  idDrink: '14378',
  strDrink: 'Lunch Box',
  strCategory: 'Beer',
  strAlcoholic: 'Alcoholic',
  strGlass: 'Pint glass',
  strInstructions: 'Fill a pint glass almost full with beer. Then fill the rest with orange juice (careful not to fill it to the top). Then take the shot of Amaretto and drop it in.',
  strInstructionsDE: 'Füllen Sie ein Pintglas fast voll mit Bier. Dann den Rest mit Orangensaft füllen (darauf achten, ihn nicht nach oben zu füllen). Dann nimm den Schuss Amaretto und lass ihn rein.',
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/qywpvt1454512546.jpg',
  strIngredient1: 'Beer',
  strIngredient2: 'Amaretto',
  strIngredient3: 'Orange juice',
  strMeasure1: '3/4 bottle ',
  strMeasure2: '1 shot ',
  strMeasure3: '1 oz ',
  strCreativeCommonsConfirmed: 'No',
  dateModified: '2016-02-03 15:15:46',
};

export default CocktailListItemMock;
