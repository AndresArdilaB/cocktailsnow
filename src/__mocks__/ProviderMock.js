import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
