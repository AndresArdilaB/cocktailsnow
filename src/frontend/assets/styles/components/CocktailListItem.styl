.container_list_item 
  width: 254px
  height: 270px
  margin: 15px 10px
  border-radius: 25px
  text-align: center

.media_list_item 
  width: fit-content
  display: flex
  justify-content: center
  border: solid 2px #fff
  border-radius: 5px
  margin-right: 10px

.media_list_item img 
  width: 250px
  border-radius: 5px

.link 
  color: #fff
  text-decoration: none

.link:hover 
  text-decoration: underline
