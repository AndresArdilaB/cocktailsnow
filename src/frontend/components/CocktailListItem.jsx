import React from 'react';
import { Link } from 'react-router-dom';
import '../assets/styles/components/CocktailListItem.styl';

const CocktailListItem = ({ idDrink, strDrink, strDrinkThumb }) => {

  return (
    <div className='container_list_item'>
      <Link to={`/cocktail/${idDrink}`} className='link'>
        <div className='media_list_item'>
          <img src={strDrinkThumb} alt='The Cocktel' />
        </div>
        <div>
          {strDrink}
        </div>
      </Link>
    </div>
  );
};

export default CocktailListItem;
