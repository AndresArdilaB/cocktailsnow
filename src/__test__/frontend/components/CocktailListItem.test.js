/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import CocktailListItemMock from '../../../__mocks__/CocktailListItemMock';
import CocktailListItem from '../../../frontend/components/CocktailListItem';

describe('<CocktailListItem />', () => {

  test('Component render', () => {
    const cocktailListItem = shallow(
      <ProviderMock>
        <CocktailListItem />
      </ProviderMock>,
    );

    expect(cocktailListItem.length).toEqual(1);
  });

  test('Test cocktail detail', () => {
    const wrapper = mount(
      <ProviderMock>
        <CocktailListItem
          {...CocktailListItemMock}
        />
      </ProviderMock>,
    );

    wrapper.find('Link').simulate('click');
    expect(wrapper.find('Link').at(0).props().to).toBe(`/cocktail/${CocktailListItemMock.idDrink}`);
  });

});

describe('CockailListItem Snapshot', () => {
  test('Testiong render component UI', () => {
    const cocktailListItem = create(
      <ProviderMock>
        <CocktailListItem />
      </ProviderMock>,
    );

    expect(cocktailListItem.toJSON()).toMatchSnapshot();
  });
});
