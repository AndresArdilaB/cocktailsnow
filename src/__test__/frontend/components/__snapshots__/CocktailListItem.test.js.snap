// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CockailListItem Snapshot Testiong render component UI 1`] = `
<div
  className="container_list_item"
>
  <a
    className="link"
    href="/cocktail/undefined"
    onClick={[Function]}
  >
    <div
      className="media_list_item"
    >
      <img
        alt="The Cocktel"
      />
    </div>
    <div />
  </a>
</div>
`;
